### PR TITLE
wifi: Display rts threshold in wifi shell

### PR DESCRIPTION
**Pull request by kapbh**
Add a new offload API to get rts threshold and
display to the user in wifi shell.

### CODE REVIEW

File: include/zephyr/net/wifi_mgmt.h [hunk 1]
```diff
@@ -105,8 +105,10 @@ enum net_request_wifi_cmd {
 	/** Set enterprise mode credential */
 	NET_REQUEST_WIFI_CMD_ENTERPRISE_CREDS,
 /** @cond INTERNAL_HIDDEN */
-	NET_REQUEST_WIFI_CMD_MAX
+	NET_REQUEST_WIFI_CMD_MAX,
 /** @endcond */
+	/** Get RTS threshold */
+	NET_REQUEST_WIFI_CMD_RTS_THRESHOLD_CONFIG
```
Comment (krish2718) [inline include/zephyr/net/wifi_mgmt.h:111]: This have to be added before MAX

File: subsys/net/l2/wifi/wifi_shell.c [hunk 2]
```diff
@@ -1695,28 +1695,49 @@ static int cmd_wifi_set_rts_threshold(const struct shell *sh, size_t argc, char
...
+		}
+
+		if (net_mgmt(NET_REQUEST_WIFI_RTS_THRESHOLD, iface,
+			     &rts_threshold, sizeof(rts_threshold))) {
+			shell_fprintf(sh, SHELL_WARNING,
+					"Setting RTS threshold failed.\n");
+			return -ENOEXEC;
+		}
+
+		if ((int)rts_threshold >= 0) {
```
Comment (krish2718) [inline subsys/net/l2/wifi/wifi_shell.c:1736]: Isn't `== 0` off?

File: subsys/net/l2/wifi/wifi_shell.c [hunk 3]
```diff
@@ -1695,28 +1695,49 @@ static int cmd_wifi_set_rts_threshold(const struct shell *sh, size_t argc, char
...
-		long rts_val = shell_strtol(argv[1], 10, &err);
+	if (argc > 2) {
+		PR_WARNING("Invalid number of arguments\n");
+		return -ENOEXEC;
+	}
 
-		if (err) {
-			shell_error(sh, "Unable to parse input (err %d)", err);
-			return err;
+	if (argc == 1) {
```
Comment (krish2718) [inline subsys/net/l2/wifi/wifi_shell.c:1703]: may be add support for `show`?

**Review by krish2718 — CHANGES_REQUESTED** [review]

File: subsys/net/l2/wifi/wifi_shell.c (hunk 2 above)
Comment (kapbh) [reply subsys/net/l2/wifi/wifi_shell.c:1736]: The range of rts threshold is between 0-2347 bytes. The value -1 will suggest that it's off.

File: subsys/net/l2/wifi/wifi_shell.c (hunk 3 above)
Comment (kapbh) [reply subsys/net/l2/wifi/wifi_shell.c:1703]: Ok will add like `wifi rts_threshold show`

Comment (kapbh) [reply subsys/net/l2/wifi/wifi_shell.c:1703]: @krish2718 , as we are adding `show`. I think we can add `wifi rts_threshold set value/off` to setting value. Right?

Comment (krish2718) [reply subsys/net/l2/wifi/wifi_shell.c:1703]: Either way works for me, we can leave it as `no_args == show` and `arg == value`.

**Review by krish2718 — APPROVED** [review]

**Review by jukkar — APPROVED** [review]

**Pull request merged**